src/ImageSharp/Compression/Zlib/DeflaterEngine.cs

```diff
@@ -486,15 +486,16 @@ private bool FindLongestMatch(int curMatch)
             this.matchLen = Math.Max(this.matchLen, DeflaterConstants.MIN_MATCH - 1);
             int matchLength = this.matchLen;
 
-            if (scan + matchLength > scanMax)
+            int scanEndPosition = scan + matchLength;
+            if (scanEndPosition > scanMax)
             {
                 return false;
             }
```

Comment:
Is there a chance that `scan + matchLength` can overflow? If not, please ignore this suggestion.

To avoid potential overflow:
```suggestion
            if (scan > scanMax - matchLength)
            {
                return false;
            }
            int scanEndPosition = scan + matchLength;
```

Reply:
It should be very rare for really large files. Then the suggestion would help.